### ISSUE TITLE
3.4 pull / push issue with bitbucket (working in 3.3.18)

### ISSUE THREAD

**Issue by anton-mccoy-ts** [desktop/desktop#18730]
### The problem

The Authentication between GitHub Desktop and Bitbucket isn't working in 3.4

I reverted back to 3.3.18 and it worked once again, however this version then downloads the latest version meaning it breaks when you restart.

### Release version

3.4

### Operating system

Windows 10

### Screenshots

![gh-issue](https://github.com/desktop/desktop/assets/164172929/1eac4185-535c-4d15-aed8-a65996bad33f)

**Comment by nick-g-h** [comment]
I'm running into the same issue after updating to 3.4.0. Authentication failure with any push to or pull from my Bitbucket repos. Launching 3.3.18 and it works. 

On Windows 10 just like you.

I tried signing out of GitHub in the application and restarting it just in case, but it makes no difference.

In the meantime I am just using a shortcut to the 3.3.18: 

`C:\Users\USERNAME\AppData\Local\GitHubDesktop\app-3.3.18\GitHubDesktop.exe`

**Comment by sergiou87** [comment]
Could you upload the log file from GitHub Desktop so that I could get some more information about this error? To access the log files go to the file menu in GitHub Desktop and select Help > Show Logs.
The log files are created daily -- please upload a log file as an attachment from a day where you experienced the issue.

**Comment by anton-mccoy-ts** [comment]
2024-06-03T08:17:24.847Z - info: [ui] Stats reported.
2024-06-03T08:17:51.472Z - info: [ui] Executing getRemotes: git remote -v (took 1.503s)
2024-06-03T08:17:53.121Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 1.634s)
2024-06-03T08:17:53.121Z - info: [ui] Executing getCommits: git log a04c77d30e7e9a9a517dc471133c1006c441a27d --date=raw --max-count=1 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 1.470s)
2024-06-03T08:17:53.235Z - info: [ui] Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 -- (took 1.742s)
2024-06-03T08:17:56.185Z - info: [ui] Executing getSymbolicRef: git symbolic-ref -q refs/remotes/origin/HEAD (took 2.381s)
2024-06-03T08:17:58.277Z - info: [ui] Executing getConfigValueInPath: git config -z pull.rebase (took 2.507s)
2024-06-03T08:18:02.513Z - info: [ui] Executing getCommits: git log origin/master..master --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 4.384s)
2024-06-03T08:18:02.665Z - info: [ui] Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT (took 4.523s)
2024-06-03T08:18:02.669Z - info: [ui] Executing getStashEntries: git log -g -z --format=%gD%x00%H%x00%gs%x00%T%x00%P refs/stash (took 4.534s)
2024-06-03T08:18:04.199Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 10.498s)
2024-06-03T08:18:07.386Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 3.185s)
2024-06-03T08:18:11.025Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 2.070s)
2024-06-03T08:18:12.578Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 1.552s)
2024-06-03T08:18:14.436Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.122s)
2024-06-03T08:18:20.810Z - info: [ui] Executing getCommits: git log d2c9a137c3617eef41b1aee859b1f8ddf5706e51 --date=raw --max-count=1 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.895s)
2024-06-03T08:18:24.530Z - info: [ui] Executing getRemotes: git remote -v (took 3.708s)
2024-06-03T08:18:31.646Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 7.110s)
2024-06-03T08:18:33.003Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 1.356s)
2024-06-03T08:18:34.247Z - info: [ui] Executing getAheadBehind: git rev-list --left-right --count master...origin/master -- (took 1.243s)
2024-06-03T08:19:42.235Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2024-06-03T08:19:43.629Z - info: [ui] [AppStore] loading 196 repositories from store
2024-06-03T08:19:43.631Z - info: [ui] [AppStore] found account: antonmccoy76 (antonmccoy76)
2024-06-03T08:19:44.279Z - info: [ui] launching: 3.4.0 (Windows 10.0.19045)
2024-06-03T08:19:44.281Z - info: [ui] execPath: 'C:\Users\User\AppData\Local\GitHubDesktop\app-3.4.0\GitHubDesktop.exe'
2024-06-03T08:19:45.768Z - info: [ui] Executing installGlobalLFSFilter: git lfs install --skip-repo (took 1.409s)
2024-06-03T08:19:46.091Z - info: [ui] Subscribed 'antonmccoy76' to Alive channel
2024-06-03T08:19:50.438Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.607s)
2024-06-03T08:19:50.440Z - error: [ui] `git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin` exited with an unexpected code: 128.
stderr:
remote: Invalid credentials
fatal: Authentication failed for 'https://bitbucket.org/tiny-spark/clue-website.git/'

(The error was parsed as 4: Authentication failed. Some common reasons include:

- You are not logged in to your account: see File > Options.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.)
2024-06-03T08:19:50.441Z - info: [ui] askPassHandler: auth failed, deleting https://bitbucket.org/ credential

**Comment by Avoltance** [comment]
Similar issue and logs with a gitlab server.
After updating, seems like Github Desktop forgot my credentials, without my access token expiring.
But, after adding back an access token, pull / push operations worked.

```
2024-06-03T11:28:37.494Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2024-06-03T16:56:31.289Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 3.819s)
2024-06-03T16:56:35.022Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2024-06-03T16:56:36.441Z - info: [ui] [AppStore] loading 36 repositories from store
2024-06-03T16:56:36.441Z - info: [ui] [AppStore] found account: Avoltance (Avoltance)
2024-06-03T16:56:36.953Z - info: [ui] Subscribed 'Avoltance' to Alive channel
2024-06-03T16:56:36.960Z - info: [ui] launching: 3.4.0 (Windows 10.0.19045)
2024-06-03T16:56:36.960Z - info: [ui] execPath: 'C:\Users\avolt\AppData\Local\GitHubDesktop\app-3.4.0\GitHubDesktop.exe'
2024-06-03T17:04:34.012Z - error: [ui] `git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin` exited with an unexpected code: 128.
stderr:
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://git.lutes.upmc.fr/planques/flashcards.git/'

(The error was parsed as 4: Authentication failed. Some common reasons include:

- You are not logged in to your account: see File > Options.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.)
[...]
2024-06-03T17:05:01.164Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 35 repositories took 28.2s of which 0.0s paused, total 28.2s
2024-06-03T17:05:17.739Z - info: [ui] [Timing] Action 'create commit' for 'quiz-v2' took 1.401s
2024-06-03T17:05:18.573Z - info: [ui] askPassHandler: no account found for https://git.lutes.upmc.fr
2024-06-03T17:07:46.281Z - info: [ui] askPassHandler: user cancelled generic git authentication
2024-06-03T17:07:46.416Z - info: [ui] Executing push: git -c credential.helper= push origin addressable-scenes:addressable-scenes --progress (took 147.999s)
2024-06-03T17:11:29.855Z - info: [ui] askPassHandler: no account found for https://git.lutes.upmc.fr
2024-06-03T17:12:42.511Z - info: [ui] askPassHandler: acquired generic credentials for https://git.lutes.upmc.fr
2024-06-03T17:12:42.739Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 73.047s)
2024-06-03T17:12:48.004Z - info: [ui] Executing push: git -c credential.helper= push origin addressable-scenes:addressable-scenes --progress (took 3.366s)
```

**Comment by MaxEtMoritz** [comment]
Same issue for me, GitHub Desktop v3.4.0 forgot my credentials for the university's GitLab Server.
I checked the contents of Windows credential manager after re-entering the credentials.
The old entry `GitHub - git.my-university.de/my-username` is still there, but somehow GitHub Desktop now created a new entry `GitHub - https://git.my-university.de//my-username` with https:// at the beginning and an additional slash between username and url.

If this change in behavior was intentional it would be good to migrate old credentials to the new format if possible.

**Comment by sergiou87** [comment]
Yes, if you're prompted for the credentials you have to enter them again, just providing them should be enough. You can recover them from the macOS Keychain or the Windows Credential Manager as @MaxEtMoritz suggested.

@MaxEtMoritz it's a bit more complicated than that because the URL used in old entries wasn't always correct 🤦‍♂ 

That said, I'm surprised of that `//my-username` in the new URL. I don't think that should happen and I don't see it in the entries GitHub Desktop created for me.

**Comment by Avoltance** [comment]
I can confirm a new entry in Windows Credential Manager has been made with an additonal slash.

![image](https://github.com/desktop/desktop/assets/14175076/4fa2756b-02b2-436e-850a-853401fbd9f2)

**Comment by niik** [comment]
:wave: we just shipped GitHub Desktop 3.4.2-beta2 to our [beta channel](https://desktop.github.com/beta/). This includes support for using [Git Credential Manager](https://gh.io/gcm). It's currently off by default but you can enable it in Settings under the advanced tab. We believe this will address the authentication problems outlined in this issue.

<img width="608" alt="Screenshot of Desktop's advanced preferences section with a highlight around the new option to enable Git Credential Manager" src="https://github.com/desktop/desktop/assets/634063/5d4da983-7663-49dd-bb7b-ce51ee9d2d3a" />

When Git Credential Manager (GCM) is enabled all credentials for third-party (i.e. Azure Devops, BitBucket, GitLab etc) will be handled, and stored, outside of GitHub Desktop. GCM supports browser-sign in to many popular services and will avoid the need to create personal access tokens (PATs). 

Please download our beta version from https://desktop.github.com/beta/ to give this a go and please report back with your experience. We'll be collecting feedback in our beta channel which will inform when we make this available to all users. Installing the beta version will not clear your preferences or tracked repositories and you can switch to the production version by simply downloading the latest version from https://desktop.github.com.

**Comment by antonmccoy76** [comment]
Sorry for not getting back to you. This fix has resolved this issue. Thank you for the time and effort you have put into looking into this for me (and others).
